### PR TITLE
Client's history clear alternative

### PR DESCRIPTION
**Pull request by takman1**
Another way to clear client's history.

### CODE REVIEW

**Review by javiereguiluz — COMMENTED** [review]
@takman1 thanks for this contribution!

We generally try to avoid explaining multiple ways of doing the same thing. The problem is that doing it makes users wonder which is the best way and which one should they use in their apps.

So, about this proposal: are `restart()` and `getHistory()->clear()` equivalent? Does restart() do more things?

Depending on the answers we should:

* Remove restart() and only explain clear()
* Reword the explanation of restart() and add clear() as something different
* Keep restart() and not document clear()

Thanks!

**Comment by takman1** [comment]
Hello @javiereguiluz and thanks for your reply.

I get your point on over explaining things.
The point is that ``restart()`` method does 2 things : 
 - Clearing the history 
 - Removing all cookies.

IMHO I think we should have 2 examples 1 for each case (history and cookies). The reader gets whta's under the hood.
But keeping ``restart()`` method is also fine as it keeps things simple.

**Comment by javiereguiluz** [comment]
I agree with you. We could document this `clear()` method and change the existing doc for `restart()`. Something like this:

```php
// -- Before --------------------------

// delete history
$client->restart();

// -- After ---------------------------

// reinitializes the browser state (history and cookies are deleted)
$client->restart();
```

File: components/browser_kit.rst [hunk 1]
```diff
@@ -222,8 +222,9 @@ also delete all the cookies::
     $client = new Client();
     $client->request('GET', 'http://symfony.com');
 
-    // delete history
+    // reinitializes the browser state (history and cookies are deleted)
     $client->restart();
+    // which internally calls history->clear() and cookieJar->clear()
```
Comment (javiereguiluz) [inline components/browser_kit.rst]: Not sure about this line:

```
which internally calls history->clear() and cookieJar->clear()
```

It explains an internal coding details (which makes docs hard to maintain). And we already explain above that this clears cookies and history, so maybe it's unneeded. Thanks!

**Comment by javiereguiluz** [comment]
Thanks @takman1.

**Pull request closed without merging**